tests/cli/keyboard_interface.c

```diff
@@ -59,63 +53,61 @@ int connect_tcp() {
     servaddr.sin_port = htons(PORT);
 
     if ((bind(sockfd, (struct sockaddr*) &servaddr, sizeof(servaddr))) != 0) {
-        printf("socket bind failed... %s\n", strerror(errno));
+        printf("Keyboard interface: socket bind failed... %s\n", strerror(errno));
         exit(0);
     }
 
-    printf("Waiting for client\n");
+    printf("Keyboard interface: Keyboard waiting for client\n");
+    sleep(2);
+    
     if ((listen(sockfd, 5)) != 0) {
-        printf("listen failed...\n");
+        printf("Keyboard interface: listen failed...\n");
         exit(0);
     }
     socklen_t len = sizeof(cli);
     connfd = accept(sockfd, (struct sockaddr*) &cli, &len);
-    printf("Accepted client\n");
+    printf("Keyboard interface: Accepted client\n");
     if (connfd < 0) {
-        printf("server accept failed...\n");
+        printf("Keyboard interface: server accept failed...\n");
         exit(0);
     }
     return connfd;
 }
 
-int main() {
-    signal(SIGINT, sigint_handler);
+void setup_keyboard() {
 
     // The bitmap of buttons pressed to be sent to net handler
-    uint32_t buttons = 0;
+    uint64_t gamepad_buttons = 0;
     float joystick_vals[4] = {0};
-    // The received bitstring from Python to be parsed into BUTTONS
-    char buff[33];
-
-    printf("Its gamer time\n");
+    // The received gamepad bitstring from Python to be parsed into BUTTONS
+    char gamepad_buff[33];
 
-    printf("Starting Net Handler CLI...\n");
-    fflush(stdout);
-
-    start_net_handler();
-    sleep(2);
-    // Connect dummy gamepad and start teleop mode
-    send_user_input(0, joystick_vals, GAMEPAD);
-    send_run_mode(DAWN, TELEOP);
-    sleep(2);
+    uint64_t keyboard_buttons = 0;
+    // The received keyboard bitstring from Python
+    char keyboard_buff[48];
```

Comment:
In `keyboard_inputs.py`, is the gamepad bitmap 32 bits or `NUM_GAMEPAD_BUTTONS` bits when sent over the socket?

It seems to me that `char gamepad_buff[33]` is based off of 32-bits, while `char keyboard_buff[48]` is based off of `NUM_KEYBOARD_BUTTONS`. We should keep them consistent!

Reply:
Can you explain what you mean? They are 33 and 48 to keep account of the NULL terminator at the end, if you mean that I should use the defined `NUM_GAMEPAD_BUTTONS` + 1 and `NUM_KEYBOARD_BUTTONS`+1 then yea I can do that

Reply:
Ok I see. Yeah we should use `#define`'s where possible. I was confused because `NUM_GAMEPAD_BUTTONS` is 17. With the joysticks as possible "buttons", there still aren't 32 buttons from my understanding. I wasn't sure where `32 + 1` came from